internal/epoch/epoch_utils.go

```diff
@@ -91,3 +93,57 @@ func deletionWatermarkFromBlobID(blobID blob.ID) (time.Time, bool) {
 
 	return time.Unix(unixSeconds, 0), true
 }
+
+type intRange struct {
+	lo, hi int
+}
+
+func (r intRange) length() uint {
+	return uint(r.hi - r.lo)
+}
+
+func (r intRange) isEmpty() bool {
+	return r.length() == 0
+}
+
+var errNonContiguousRange = errors.New("non-contiguous range")
+
+// constants from the standard math package.
+const (
+	//nolint:gomnd
+	intSize = 32 << (^uint(0) >> 63) // 32 or 64
+
+	maxInt = 1<<(intSize-1) - 1
+	minInt = -1 << (intSize - 1)
+)
+
+// Returns a continuous close-open epoch range for the keys, that is [lo, hi).
+// A range of the form [v,v) means the range is empty.
+// When the range is not continuous an error is returned.
+func getKeyRange[E any](m map[int]E) (intRange, error) {
+	var count uint
```

Comment:
Is there a specific reason to use `uint` here? The built-in `len` returns `int`.